### PR TITLE
Add it tests for main.rs and test coverage metrics

### PR DESCRIPTION
**Pull request by javierpedreira**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
+sudo: required
 language: rust
+cache:
+  - cargo
+  - directories:
+      - "${HOME}/kcov/"
```
Comment (nathankleyn) [inline .travis.yml]: This doesn't override the usual directories does it? Is it additive?

File: .travis.yml [hunk 2]
```diff
@@ -8,6 +13,13 @@ addons:
     packages:
       # For building MUSL static builds on Linux.
       - musl-tools
+      - cmake
```
Comment (nathankleyn) [inline .travis.yml]: Are these new packages all for `cargo-travis`?

File: server/src/main.rs [hunk 3]
```diff
@@ -121,3 +121,105 @@ fn parse_args<'a, 'b>() -> clap::App<'a, 'b> {
...
+    }
+
+    #[test]
+    #[should_panic]
+    fn run_server_with_invalid_port() {
+        run_server(DEFAULT_HOST, "", Path::new(DEFAULT_REPO_ROOT));
+    }
+
+    #[test]
+    #[should_panic]
```
Comment (nathankleyn) [inline server/src/main.rs]: Nice work! These annotations are awesome and never get used enough by people! 👏

File: server/src/main.rs [hunk 4]
```diff
@@ -121,3 +121,105 @@ fn parse_args<'a, 'b>() -> clap::App<'a, 'b> {
...
+        .start(|app| {
+            app.resource("/repo/{repo}", |r| r.h(&super::get_repo));
+        })
+    }
+
+    #[test]
+    fn get_repo_with_file_in_root_folder() {
+        let mut srv = test_server();
+
+        let query = format!("/repo/{}?file={}&reference={}", "", "README.md", "heads/master");
```
Comment (nathankleyn) [inline server/src/main.rs]: These are all interpolating static strings into another static string — best to just have one static string in thef first place?

```suggestion
        let query = "/repo/?file=README.md&reference=heads/master";
```

Comment (nathankleyn) [reply server/src/main.rs]: Same for all the other `format!` calls in here.

File: .travis.yml (hunk 2 above)
Comment (javierpedreira) [reply .travis.yml]: Yes they are.

File: .travis.yml [hunk 5]
```diff
@@ -1,13 +1,24 @@
+sudo: required
```
[the remark this answers is not in the document]
Comment (javierpedreira) [reply .travis.yml]: Ah tha'd be because I picked and old tutorial? I'll try removing it

File: .travis.yml (hunk 1 above)
Comment (javierpedreira) [reply .travis.yml]: As we spoke, it is an arbitrary dir cache for kcov

**Pull request merged**